vehicle/mkvi/software/sensing/sensing.c

```diff
@@ -0,0 +1,73 @@
+/*
+Header:
+    Code for the Sensing Controller Board located in the nostril motherboard and
+the rear sensing box Author:
+    @author Richard Li
+    @date   03/25/2023
+*/
+
+/*----- Includes -----*/
+#include "sensing.h"
+#include "can_api.h"
+#include "log_uart.h"
+#include "utils/timer.h"
+#include <avr/interrupt.h>
+#include <avr/io.h>
+#include <stdio.h>
+#include <string.h>
+#include <util/delay.h>
+
+volatile uint8_t gTimerFlag = 0x01;
+uint8_t gStatusMessage[8] = {};
+volatile uint8_t can_recv_msg[8] = {};
+
+void timer1_isr(void) {
+    send_can = true;
+};
+
+int pcint0_callback(int counter_left) {
+    // Left wheel speed sensor is in pin PB2 -> PORTB
+    sensing.wsl = !gpio_get_pin(WHEEL_SPEED_LEFT);
```

Comment:
nit: You use `ws1` for the left sensor, but `wsr` for the right. Make sure you keep your naming convention consistent, l and r make sense to me.